anaconda_packaging_utils/api/repodata_api.py

```diff
@@ -0,0 +1,375 @@
+        Architecture.LINUX_X86_32,
+        Architecture.LINUX_ARM_V6L,
+        Architecture.LINUX_ARM_V7L,
+        Architecture.LINUX_PPC64LE,
+        Architecture.OSX_X86_64,
+        Architecture.OSX_X86_32,
+        Architecture.WIN_64,
+        Architecture.WIN_32,
+        Architecture.NO_ARCH,
+    },
+    Channel.MSYS_2: {
+        Architecture.WIN_64,
+        Architecture.WIN_32,
+    },
+}
+
+
+class ApiException(BaseApiException):
+    """
+    Indicates an exception occurred with this API.
+    """
+
+    pass
+
+
+@dataclass
+class RepodataMetadata:
+    """
+    Metadata section in a `repodata.json` blob
+    """
+
+    ## Required fields ##
+    subdir: str
+    ## Optional fields ##
+    arch: Optional[str] = None
+    platform: Optional[str] = None
+
+
+@dataclass
+class PackageData:
+    """
+    Per-package data stored in a `repodata.json` blob
+    """
+
+    ## Required fields ##
+    build: str
+    build_number: int
+    depends: list[str]
+    md5: str
+    sha256: str
+    name: str
+    size: int
+    version: str
+    subdir: str
+    ## Optional fields ##
+    timestamp: Optional[int] = None
+    date: Optional[str] = None
+    track_features: Optional[str] = None
+    license: Optional[str] = None
+    license_family: Optional[str] = None
+
+    # TODO rm: Enforce type checking when this PR is released
+    #   https://github.com/conda/conda/pull/13385
+    @no_type_check
+    def __lt__(self, other: object) -> bool:
+        """
+        Allows for version comparisons between two `PackageData` instances
+        NOTE: We are relying on the default implementation of `__eq__()` for dataclasses, which checks that all fields
+              are equivalent.
+        NOTE: Type checking is disabled for this function as `VersionOrder`
+        :param other: Object to compare against this one.
+        :returns: If this instance has a lower version than the compared instance.
+        """
+        if not isinstance(other, PackageData):
+            return False
+        # Comparing versions between two different software projects/packages is meaningless.
+        if self.name != other.name:
+            return False
+        try:
+            self_v = VersionOrder(self.version)
+            other_v = VersionOrder(other.version)
+            # If two versions are equal, fallback to comparing the build number
+            if self_v == other_v:
+                return self.build_number < other.build_number
+            return self_v < other_v
+        except InvalidVersionSpec:
+            # If a `VersionOrder` object failed to be created, we are not able to accurately compare these objects.
+            pass
+        return False
+
+
+@dataclass
+class Repodata:
+    """
+    Structure that contains an entire serialized `repodata.json` blob
+    """
+
+    info: RepodataMetadata
+    packages: dict[str, PackageData]
+    removed: list[str]
+    repodata_version: int
+
+
+_REPODATA_JSON_SCHEMA: Final[SchemaType] = {
```

Comment:
Is this derived from https://github.com/conda/schemas?

Reply:
Don't know how I missed this comment initially. No it isn't, I didn't know this existed. I reverse engineered this schema from examples and then ran an automated test against multiple real world examples.

There is a new schema in the latest code for `channeldata.json` but I only validate the `subdirs` field I use to look-up supported architectures.